core/src/main/scala/tamer/Tamer.scala

```diff
@@ -14,16 +13,15 @@ import zio.kafka.admin.{AdminClient, AdminClientSettings}
 import zio.kafka.consumer.Consumer.{AutoOffsetStrategy, OffsetRetrieval}
 import zio.kafka.consumer._
 import zio.kafka.producer.{Producer, ProducerSettings}
-import zio.kafka.serde.{Deserializer, Serializer}
+import zio.kafka.serde.{Serde => ZSerde, Serializer}
 import zio.stream.{UStream, ZStream}
 
 trait Tamer {
   def runLoop: ZIO[Clock, TamerError, Unit]
 }
 
 object Tamer {
-  private[tamer] final case class StateKey(stateKey: String, groupId: String)
-  private[this] final val stateKeySerde = Serde.key[StateKey]
+  final case class StateKey(stateKey: String, groupId: String)
```

Comment:
visibility needs to be public as `Codec[Tamer.StateKey]` needs to be provided by end user now